cpp/pipelines/add_authority_external_ref.cc

```diff
@@ -130,19 +181,164 @@ void ParseGndWikidataMappingFile(std::string filename,
+                    top_level_number = -1;
+                    second_level_number = -1;
+                    is_start_group = false;
+                    gnd_id_temp_string = "";
+                    gnd_data = {};
+                }
+            }
+
+            if (is_start_group) {
+                // std::cout << top_level_number << " , " << second_level_number << std::endl;
+                if (line_parsed[0][0].get<int>() == top_level_number && line_parsed[0][1].get<int>() == second_level_number) {
+                    if (!line_parsed[1].is_structured()) {
+                        if (line_parsed[1].is_string()) {
+                            if (DoesItMatch(wikipedia_address, line_parsed[1].get<std::string>())) {
+                                gnd_data.wikipedia_personal_address = line_parsed[1].get<std::string>();
+
+                                ++total_number_of_wikipedia;
+                            }
+
+                            // if wikidata
+                            if (DoesItMatch(wikidata_address, line_parsed[1].get<std::string>())) {
+                                wikidata_temp_string = nlohmann::to_string(line_parsed[1]);
+                                gnd_data.wikidata_personal_entity_id = wikidata_temp_string.substr(
+                                    wikidata_address_str_lenght + 1, (wikidata_temp_string.length() - (wikidata_address_str_lenght + 2)));
+
+                                ++total_number_of_wikidata;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        std::cout << "\r"
+                  << "Parsed: " << total_line_parsed << " line(s), "
+                  << " Total GND-ID: " << total_numbers_of_gnd_id_generated << ", Total GND with Wikidata: " << total_number_of_wikidata
+                  << ", Total GND with Wikipedia: " << total_number_of_wikipedia;
+        //   << "GND-ID url: " << tmp_id;
+        std::cout.flush();
+        ++total_line_parsed;
+    }
+
+    const auto load_file_end(std::chrono::high_resolution_clock::now());
+    auto duration = std::chrono::duration_cast<std::chrono::seconds>(load_file_end - load_file_start);
+
+    std::cout << std::endl
+              << "Total GND-ID: " << total_numbers_of_gnd_id_generated << std::endl
+              << "Total GND with Wikidata: " << total_number_of_wikidata << std::endl
+              << "Total GND with Wikipedia: " << total_number_of_wikipedia << std::endl;
+    std::cout << "Total time of computation: " << duration.count() << " second(s)" << std::endl;
+
+    input_file.close();
+    return true;
+}
+
+int Main(int argc, char *argv[]) {
+    if (argc < 4)
         Usage();
 
     const std::string marc_input_filename_or_create_flag(argv[1]);
     const std::string marc_output_filename_or_dnb_input(argv[2]);
     const std::string mapping_txt_filename(argv[3]);
 
+    // if (marc_input_filename_or_create_flag == "--create_mapping_file") {
```

Comment:
Please remove this block of old commented out code entirely. If somebody wants to have a look at the old code, he can still look into the git history.